src/modules/decklink/producer/decklink_producer.cpp

```diff
@@ -382,10 +424,30 @@ class decklink_producer : public IDeckLinkInputCallback
     virtual ULONG STDMETHODCALLTYPE AddRef() { return 1; }
     virtual ULONG STDMETHODCALLTYPE Release() { return 1; }
 
-    virtual HRESULT STDMETHODCALLTYPE VideoInputFormatChanged(BMDVideoInputFormatChangedEvents /*notificationEvents*/,
-                                                              IDeckLinkDisplayMode* newDisplayMode,
+    virtual HRESULT STDMETHODCALLTYPE VideoInputFormatChanged(BMDVideoInputFormatChangedEvents notificationEvents,
+                                                              IDeckLinkDisplayMode*            newDisplayMode,
                                                               BMDDetectedVideoInputFormatFlags /*detectedSignalFlags*/)
     {
+        auto mode = newDisplayMode->GetDisplayMode();
+        auto fmt     = get_caspar_video_format(mode);
+
+        auto new_fmt     = core::video_format_desc(fmt);
+
+        CASPAR_LOG(info) << print() << L" Input format changed from " << input_format.name << L" to " << new_fmt.name;
+
+        input_format = new_fmt;
+
+        input_->PauseStreams();
+
+        // reinitializing filters because not all filters can handle on-the-fly format changes
+        video_filter_ = Filter(vfilter_, AVMEDIA_TYPE_VIDEO, format_desc_, mode_);
+        audio_filter_ = Filter(afilter_, AVMEDIA_TYPE_AUDIO, format_desc_, mode_);
+
+        // reinitializing video input with the new display mode
+        input_->EnableVideoInput(
```

Comment:
Does audio still work after this with no call to `input_->EnableAudioInput`?

Reply:
Yes it does, I've double checked and it works. No need for `EnableAudioInput`.